mypy/types.py

```diff
@@ -2356,15 +2360,38 @@ def visit_instance(self, t: Instance) -> str:
                 s += '[{}]'.format(self.list_str(t.args))
         if self.id_mapper:
             s += '<{}>'.format(self.id_mapper.id(t.type))
+        if mypy.options._based:
+            if s.startswith("builtins."):
+                return s.partition(".")[2]
```

Comment:
this logic is duplicated here, can it be turned into a function https://github.com/KotlinIsland/basedmypy/pull/290/files#diff-81ed300e83205366be4a93b57bbe65edb0826dbeda7f1655e506d5cc103c69c0R1701-R1703

Reply:
The other one was removed, still think it's good as a func?

Reply:
i dont think it needs to be a func if theres one instance, but i still see the other one